src/uptane/tufrepository.cc

```diff
@@ -183,15 +183,6 @@ void TufRepository::initRoot(const Json::Value& new_content) {
                                                                << requiredThreshold);
       throw IllegalThreshold(name_, "root.json contains a role that requires too many signatures");
     }
-    for (Json::ValueIterator it_keyid = (*it)["keyids"].begin(); it_keyid != (*it)["keyids"].end(); ++it_keyid) {
```

Comment:
This breaks the spec. The TUF spec specifically says that key IDs *MUST* be calculated locally which means that they also *MUST* match the key IDs calculated by the server.